### ISSUE TITLE
Newtonsoft.JSon.Schema 3.0.1 return error when using at azure function

### ISSUE THREAD

**Issue by nintexkj** [JamesNK/Newtonsoft.Json.Schema#97]
Hi I found an issue when I try to return a JSonSchema in my azure function, it returns:
**"Unable to cast object of type 'Newtonsoft.Json.Schema.Infrastructure.Collections.JSchemaPatternDictionary' to type 'System.Collections.IDictionary'."**

It happen on JSonSchema: 3.0.2 and Newtonsoft.Json: 10.0.3
I tried the same code on JSonSchema: 2.0.10 and Newtonsoft.Json: 9.0.1 it is working fine

Here is the sample code:

```
using System.Net;
using Newtonsoft.Json.Schema;

public static async Task<HttpResponseMessage> Run(HttpRequestMessage req, TraceWriter log)
{
    log.Info("C# HTTP trigger function processed a request.");

    // parse query parameter
    string name = req.GetQueryNameValuePairs()
        .FirstOrDefault(q => string.Compare(q.Key, "name", true) == 0)
        .Value;

    // Get request body
    dynamic data = await req.Content.ReadAsAsync<object>();

    // Set name to query string or body data
    name = name ?? data?.name;

    JSchema schema = new JSchema
    {
    };

    return req.CreateResponse(System.Net.HttpStatusCode.OK, schema); 
}

```

**Comment by JamesNK** [comment]
Are these the lines that cause it to fail?

```
    JSchema schema = new JSchema
    {
    };

    return req.CreateResponse(System.Net.HttpStatusCode.OK, schema); 
```

**Comment by JamesNK** [comment]
Can you include the full stacktrace of the error?

**Comment by JamesNK** [comment]
I believe this has been fixed with 3.0.3 https://www.nuget.org/packages/Newtonsoft.Json.Schema/3.0.3
